uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -341,24 +354,32 @@ define([
        * Also, checks if its an update request from manager and accordingly updates leave request
        */
       this.submit = function () {
```

Comment:
Implemented the same but one change,
```javascript
this.submitting = true;
changeStatusBeforeSave.call(this);
```
is moved before calling `validateBeforeSubmit` as it involves API call, so submitting needs to be set, also before calling `isValid`API, the status with which the request will be saved needs to be set.